ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -507,6 +509,25 @@ KEXT_STATIC int HandleVnodeOperation(
                 {
                     goto CleanupAndReturn;
                 }
+                
+                if (s_osSupportsRenameDetection)
+                {
+                    PrjFSDirectoryXAttrData directoryXattr = {};
+                    SizeOrError xattrResult = Vnode_ReadXattr(currentVnode, PrjFSDirectoryXAttrName, &directoryXattr, sizeof(directoryXattr));
+                    if (xattrResult.error == 0)
+                    {
+                        if (xattrResult.size == sizeof(directoryXattr))
```

Comment:
Could it be the same size and still not be the same attrib?  

Wondering for this whole block if just this would suffice
```
PrjFSDirectoryXAttrData directoryXattr = {}; 
SizeOrError xattrResult = Vnode_ReadXattr(currentVnode, PrjFSDirectoryXAttrName, &directoryXattr, sizeof(directoryXattr));
if (xattrResult.errror != ENOATTR)
{
   kauthResult = KAUTH_RESULT_DENY;
   goto CleanupAndReturn;
}
```